### ISSUE TITLE
Deploy 'External Vocabulary' scripts, developed by QDR, to Author and Funding fields on Harvard Dataverse

### ISSUE THREAD

**Issue by cmbz** [IQSS/dataverse.harvard.edu#326]
## Overview
- ORCID and ROR 'External Vocabulary' scripts developed by QDR have been tested on the Author Name and Author Affiliation fields on demo.dataverse.org, as well as the Funding Information Agency field.
- Following review by members of the UX WG and conversation with Dataverse team members we have decided to deploy the scripts on Harvard Dataverse in order to support our work to support the GREI Metadata Recommendations.
  - See: [GREI Metadata and Search Subcommittee Recommendations_V01_2023-06-29](https://zenodo.org/records/8101957)
- The purpose of this issue is to deploy the 'External Vocabulary' scripts on Harvard Dataverse, with the watermarks changed to read "Select or enter...":

  <kbd><img width="948" alt="Screenshot 2024-12-10 at 5 16 35 PM" src="https://github.com/user-attachments/assets/2df3cc66-ff96-4979-9058-579ae7baa655"></kbd>

  <kbd><img width="952" alt="Screenshot 2024-12-10 at 5 16 40 PM" src="https://github.com/user-attachments/assets/9057bcb7-9e93-4975-bf23-06371867a969"></kbd>

See [the rationale for the watermark change](https://github.com/IQSS/dataverse.harvard.edu/issues/326#issuecomment-2528895775).

## Related
- [Review of changes to the Author and Funding Information fields on Demo Dataverse](https://docs.google.com/document/d/1tBPsWwRrWTHYwgP38eaTwJd_d70rL0WQ65xflkABeCM/edit?usp=sharing)
- [Running meeting agendas and notes for Dataverse UX Working Group](https://docs.google.com/document/d/1Ir5IcqCSdzlBGLZA_Ruh6CpR8JsUgEK5bpg-UBgmWlo/edit?usp=sharing)
- [Running Notes: Metadata Coordination](https://docs.google.com/document/d/1PGQpI_9NbGxHLl6XLXtfEVWlqriTuxLzYsMSbltaUSg/edit?usp=sharing)
- https://github.com/IQSS/dataverse-pm/issues/127
- https://github.com/IQSS/dataverse/issues/11075

**Comment by jggautier** [comment]
Hey @qqmyers. 

> Can you pick a custom watermark from the options you gave?

For everyone else, here's what I wrote from today's meeting ([see the Google Doc meeting notes for more](https://docs.google.com/document/d/1PGQpI_9NbGxHLl6XLXtfEVWlqriTuxLzYsMSbltaUSg/edit?tab=t.0)):
1. Using the text "Enter a person or organization" would keep the text short, since we've got limited space
2. Using the text "Select or enter a person or organization", although some of this text is always cut off, regardless of the width of the field
3. Using the text "Select or enter...". This might be more in line with the watermarks/placeholders used for other fields
4. Removing the watermark/placeholder

Of those options I gave, I'd pick the third for now, "Select or enter...". It's the shortest of those 3 options; by not mentioning people organizations, I think it addresses the potential issue of users thinking that they can add only people; and it follows the "Select..." watermarks/placeholders of other fields that let depositors choose from lists, like the Identifier Type fields.

> Also - we focused on the author fields - should we deploy on other fields as well?

I agree during today's meeting we talked mostly about the Author fields. In other meetings, we talked about the Author and Funding Information fields, those are the two fields whose changes I reviewed, and Sherry reviewed that review.

We haven't talked about and I haven't reviewed the changes I see on the Depositor field on Demo Dataverse. I'm not sure why I missed that that field was changed, too. I don't think we should deploy on the Depositor field or on other fields for now. I wrote that after evaluation, we can use what we learn to adjust the redesign of the Author and Funding Information fields and redesign other fields used to describe people and organizations, which would include the Depositor field.

@cmbz, I'd like to remove the mention of the Depositor field from your description. Would that be okay?

**Comment by jggautier** [comment]
Thanks for removing mention of the Depositor field @cmbz!

@qqmyers I just edited what I wrote about the watermarks/placeholder text so that hopefully the reasoning is clearer. 

@cmbz adjusted [her comment](https://github.com/IQSS/dataverse.harvard.edu/issues/326#issue-2727641037) to point to what I wrote about how the watermarks/placeholder text should be changed. So I'm assuming "Select or enter..." will be used when Harvard Dataverse's Author and Funding Information fields are changed.

@cmbz, can we change the watermarks/placeholder text that we see on Demo Dataverse, too? That'll make it possible to use Demo Dataverse for the usability testing I'm planning now.

**Comment by pdurbin** [comment]
Do we want feedback from the community on the proposed changes? I have a link to this issue queued up in the [draft](https://docs.google.com/document/d/1DCPNWXssNVJCPlg-eckI7J8Blc5XWBRXVQSPitFCWOY/edit?usp=sharing) of the next community news, but I can remove it if it's premature.

**Comment by cmbz** [comment]
@jggautier please see: https://github.com/IQSS/dataverse.harvard.edu/issues/327

**Comment by jggautier** [comment]
> Do we want feedback from the community on the proposed changes? I have a link to this issue queued up in the [draft](https://docs.google.com/document/d/1DCPNWXssNVJCPlg-eckI7J8Blc5XWBRXVQSPitFCWOY/edit?usp=sharing) of the next community news, but I can remove it if it's premature.

Definitely, requesting feedback in this way is great in general. And since the changes will also be on Demo Dataverse, maybe the community news post can point to Demo Dataverse to see how the two fields are being changed?

Also, I know this is sort of semantics, but writing that the "citation metadata block is being updated" might make some folks think that the citaion.tsv file is being changed. I know that we know that that's not the case here, so I think it's okay that that's what this GitHub issue's title reads.

I'll be reaching to different folks in the community for usability testing, so maybe the community news post can ask folks to contact me if they're interested in usability testing, too?

**Comment by jggautier** [comment]
Thanks for creating that GitHub issue @cmbz!

**Comment by cmbz** [comment]
@jggautier please rename the issues, as needed, for clarity if the titles cause confusion.

**Comment by pdurbin** [comment]
I was confused and thought citation.tsv was changing. It sounds like we're changing the Javascript instead. Something in https://github.com/gdcc/dataverse-external-vocab-support I assume (not sure which PR). For now I'll leave that bullet in the news to remind me to circle back to this issue. Perhaps we start with an announcement to the google group to check out the new feature on demo. Then from the news, we link to that announcement. Something like that.

**Comment by jggautier** [comment]
I adjusted the titles and descriptions of this GitHub issue and the issue at https://github.com/IQSS/dataverse.harvard.edu/issues/327.

As I understand it, we're adjusting the configuration of the 'External Vocabulary' scripts that have been used to change the Author and Funding Information fields on Demo Dataverse, in order to change the watermarks, then we're using that script to  change those two fields on Harvard Dataverse.

https://github.com/IQSS/dataverse.harvard.edu/issues/327 is about changing the watermarks on Demo Dataverse to match what will be on Harvard Dataverse, so that I can use Demo Dataverse for usability testing.

Your idea of using Google Groups to invite more feedback sounds good. If you don't mind, I'll can do that before next Friday, Dec 20 when we leave on break. I'd also like to ask anyone who sees that Google Groups post if they would be interested in participating in usability testing or helping recruit their users to test the changes. That way when you post the community news in early July, you could link to that announcement.

How's that sound?

**Comment by pdurbin** [comment]
The title, etc. is much better, thanks!

This issue came up in standup today and led us to add the 6.5 milestone to this PR:

- https://github.com/IQSS/dataverse/pull/11034

That is, we'll need it before we roll out the plans we have for #326 and #327.

The announcement and news plans sound good! Thanks!

**Comment by jggautier** [comment]
Hey @pdurbin. The announcement is at https://groups.google.com/g/dataverse-community/c/qZ6aQ-moFdI

**Comment by pdurbin** [comment]
@jggautier thanks! As I [mentioned](https://dataverse.zulipchat.com/#narrow/channel/434038-ux-wg/topic/design.20sprints/near/492317820) in Zulip, I have it queued up for the next community news.

**Comment by jggautier** [comment]
Ah thanks. That makes sense, since it's about things that happened the previous month. I hadn't considered that aspect of the community news posts.

I'm hoping that before next month, we'll have finished contacting folks for feedback, so using next month's community news to let folks know that we're collecting feedback shouldn't be needed. You can remove it from the queue for next months community news post.

Thanks for adding it to today's community call agenda!

**Comment by cmbz** [comment]
Related: https://github.com/gdcc/dataverse-external-vocab-support/pull/31

**Comment by sbarbosadataverse** [comment]
@ekraffmiller in today's meeting,  you mentioned creating an analysis issue - can you expand on that here?

**Comment by qqmyers** [comment]
I think https://github.com/gdcc/dataverse-external-vocab-support/blob/main/examples/authorIDandAffilationUsingORCIDandROR.md has the best install description.

**Comment by landreev** [comment]
Appears to be working, but would be great if somebody else could confirm and verify. 

![Image](https://github.com/user-attachments/assets/fde38dde-05eb-45b7-a9a7-7ed4871a9fcd)

![Image](https://github.com/user-attachments/assets/f9a88580-5c77-4d95-94f4-d5e1c5f64987)

**Comment by cmbz** [comment]
2025-05-12
- @jggautier could you please confirm this: https://github.com/IQSS/dataverse.harvard.edu/issues/326#issuecomment-2874266524 is working as expected?
- If so, please close the issue. Thanks!

**Comment by jggautier** [comment]
Hey @stevenwinship. Thanks for applying the redesign to the Author fields in Harvard Dataverse!

The redesign doesn't seem to be applied to the Funding Information fields. I checked with a new dataset I created today and with a a few datasets created last week, and those fields look the same. Is this a bug? I'm going to re-open this issue but let me know if I should create a new issue instead about these fields.

**Comment by landreev** [comment]
@jggautier I was actually working on this - I installed these control vocab. scripts in prod. Steven only closed the issue, somewhat prematurely, it appears. 
Let me take a look. 
I followed the instructions at gdcc mentioned earlier (https://github.com/gdcc/dataverse-external-vocab-support/blob/main/examples/authorIDandAffilationUsingORCIDandROR.md) - and those were just for the 2 fields, author and author affiliation. 
I am seeing the word "funding" in the title now. Let me check - there must be more parts that needed to be installed.

**Comment by landreev** [comment]
Yeah, I'm seeing it installed on demo, will replicate in prod. 
I missed it because the pr and the instructions mentioned in last few comments were all about the author parts only.

**Comment by landreev** [comment]
A quick question: on demo there's also a remote CVoc configured for the depositor field. 
Could you please confirm that this does NOT to be installed in prod.? (seems to be the case from the earlier comments here, but - ?)

**Comment by jggautier** [comment]
Yeah, the remote CVoc configured for the depositor field should not be installed on prod.

**Comment by landreev** [comment]
Please try again? 

![Image](https://github.com/user-attachments/assets/dddc71d7-fc77-4a88-b25c-5f2c2cdc9c30)

**Comment by jggautier** [comment]
Just tried it. Works how it does on Demo Dataverse. Thanks!
